doc/shared-memory-communication.md

```diff
@@ -0,0 +1,85 @@
+<!-- @todo Move the content of this file to website/getting-started/overview.md or website/for-developers/* -->
+
+# Shared memory communication
```

Comment:
This is not a new file. I used `git mv doc/conceptual-guide.md doc/shared-memory-communication.md` to rename the file. However, GitHub does not detect it as rename. Please use your favourite diff tool when reviewing.